presto-hive/src/main/java/io/prestosql/plugin/hive/HiveQlToPrestoTranslator.java

```diff
@@ -0,0 +1,169 @@
+        input = Lists.charactersOf(hiveQl).listIterator();
+    }
+
+    /**
+     * Translate a HiveQL statement to Presto SQL by fixing quoted identifiers
+     * and string literals. No other translation is included.
+     *
+     * <p>Backquotes are replaced with double quotes, including SQL-style
+     * escapes (`` becomes ` and " becomes "").
+     *
+     * <p>Single and double quotes are replaced with single quotes, with
+     * minimal processing of escape sequences to ensure that the strings end in
+     * the right place.
+     */
+    public static String translateHiveQuotesToPresto(String hiveStatement)
+    {
+        HiveQlToPrestoTranslator translator =
+                new HiveQlToPrestoTranslator(hiveStatement);
+        return translator.translateQuotedLiterals();
+    }
+
+    private static PrestoException hiveParseError(String message)
+    {
+        // TODO: This error is wrong.
+        return new PrestoException(HIVE_INVALID_VIEW_DATA,
+                format("Error translating Hive QL to Presto SQL: %s",
+                        message));
+    }
+
+    private String translateQuotedLiterals()
+    {
+        // Consume input, passing control to other translation methods when
+        // their delimiters are encountered.
+        while (input.hasNext()) {
+            char c = input.next();
+            switch (c) {
+                case '"':
+                case '\'':
+                    translateString(c);
+                    break;
+                case '`':
+                    translateQuotedIdentifier();
+                    break;
+                default:
+                    output.append(c);
+                    break;
+            }
+        }
+
+        return output.toString();
+    }
+
+    /**
+     * Start with input just after the opening backquote, consume to the
+     * closing backquote.
+     */
+    private void translateQuotedIdentifier()
+    {
+        output.append('"');
+        while (input.hasNext()) {
+            char c = input.next();
+
+            // The possible outcomes in an identifier are:
+            // - replace " with ""
+            // - leave most characters as-is
+            // - replace `` with `
+            // - replace ` with " and end the identifier
+            if (c == '"') {
+                output.append("\"\"");
+            }
+            else if (c != '`') {
+                output.append(c);
+            }
+            else if (input.hasNext() && peek() == '`') {
+                output.append('`');
+                input.next();
+            }
+            else { // end of identifier
+                output.append('"');
+                return;
+            }
+        }
+
+        throw hiveParseError("unexpected end of input in identifier");
+    }
+
+    /**
+     * Call with input just after the opening delimiter (the parameter), end
+     * with input after closing delimiter.
+     */
+    private void translateString(char delimiter)
```

Comment:
move this one method above, so methods are defined in the same order as they are used in `translateQuotedLiterals`